src/main/java/de/tum/in/www1/artemis/service/iris/session/AbstractIrisChatSessionService.java

```diff
@@ -0,0 +1,34 @@
+package de.tum.in.www1.artemis.service.iris.session;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import de.tum.in.www1.artemis.domain.iris.session.IrisChatSession;
+import de.tum.in.www1.artemis.repository.iris.IrisSessionRepository;
+
+public abstract class AbstractIrisChatSessionService<S extends IrisChatSession> implements IrisChatBasedFeatureInterface<S> {
+
+    protected IrisSessionRepository irisSessionRepository;
+
+    public AbstractIrisChatSessionService(IrisSessionRepository irisSessionRepository) {
+        this.irisSessionRepository = irisSessionRepository;
+    }
+
+    /**
+     * Updates the latest suggestions of the session.
+     *
+     * Concatenates the suggestions with "||" as separator.
+     * If a suggestion already contains "||", they are replaced with "\\||", before joining.
+     * The updated suggestions are then saved to the session in the database.
+     *
+     * @param session           The session to update
+     * @param latestSuggestions The latest suggestions to set
+     */
+    protected void updateLatestSuggestions(S session, List<String> latestSuggestions) {
+
```

Comment:
```suggestion
```

Reply:
Sorry, I couldn't quite understand what needs to be changed exactly. Thanks for reviewing

Reply:
I don't see a reason for an empty line at the top of the method

Reply:
Should be addressed now.